### PR TITLE
[Storage] Added Quick Query Parquet

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/8c1f9b739165035d9ced32761cebb19125bce233/specification/storage/data-plane/Microsoft.BlobStorage/preview/2020-06-12/blob.json
+    - C:\Users\seanmcc\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.BlobStorage\preview\2020-08-04\blob.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/autorest.md:7]: TODO fix this when https://github.com/Azure/azure-rest-api-specs/pull/13589 merges.

File: sdk/storage/Azure.Storage.Blobs/tests/Azure.Storage.Blobs.Tests.csproj [hunk 2]
```diff
@@ -23,4 +23,9 @@
     </None>
     <None Include="$(AzureStorageSharedTestSources)\azurite_cert.pem" CopyToOutputDirectory="PreserveNewest" />
   </ItemGroup>
+    <ItemGroup> 
+    <Content Include="Resources\**"> 
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory> 
+    </Content> 
+  </ItemGroup>
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Blobs/tests/Azure.Storage.Blobs.Tests.csproj:30]: Just curious what does this do?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/Azure.Storage.Blobs.Tests.csproj:30]: This says to copy the "Resources" directory with the build.  I have a parquet file that we upload as part of the Quick Query Parquet test, I put the file in Azure.Storage.Blobs.Tests/Resources.  Similar to the test files we have in Azure.StorageInternal.Avro.Tests/Resources.

File: sdk/storage/Azure.Storage.Blobs/src/Generated/Models/QueryFormat.cs [hunk 3]
```diff
@@ -23,5 +23,7 @@ public QueryFormat()
         public JsonTextConfigurationInternal JsonTextConfiguration { get; set; }
         /// <summary> arrow configuration. </summary>
         public ArrowTextConfigurationInternal ArrowConfiguration { get; set; }
+        /// <summary> Any object. </summary>
+        public object ParquetTextConfiguration { get; set; }
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Generated/Models/QueryFormat.cs:27]: how did this end up `object` ?

File: sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs [hunk 4]
```diff
@@ -57,6 +57,15 @@ internal static class QuickQueryExtensions
...
+                if (!isInput)
+                {
+                    throw new ArgumentException($"{nameof(BlobQueryParquetTextOptions)} can only be used for input serialization.");
+                }
+
+                serialization.Format.Type = QueryFormatType.Parquet;
+            }
             else
             {
                 throw new ArgumentException($"Invalid options type.  Valid options are {nameof(BlobQueryCsvTextOptions)}, {nameof(BlobQueryJsonTextOptions)}, and {nameof(BlobQueryArrowOptions)}");
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:71]: should this error message be expanded ?

File: sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs [hunk 5]
```diff
@@ -57,6 +57,15 @@ internal static class QuickQueryExtensions
                 serialization.Format.ArrowConfiguration = new ArrowTextConfigurationInternal(
                     arrowConfiguration.Schema?.Select(ToArrowFieldInternal).ToList());
             }
+            else if (textConfiguration is BlobQueryParquetTextOptions parquetTextConfiguration)
+            {
+                if (!isInput)
+                {
+                    throw new ArgumentException($"{nameof(BlobQueryParquetTextOptions)} can only be used for input serialization.");
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:64]: this isn't covered by test.

File: sdk/storage/Azure.Storage.Blobs/src/Generated/Models/QueryFormat.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Generated/Models/QueryFormat.cs:27]: The generator doesn't like it that `ParquetTextConfiguration` doesn't have any properties.  It is actually ok, `ParquetTextConfiguration` is optional if you specify `Parquet` for `QueryFormatType`.

If/when `ParquetTextConfiguration` gets additional properties, the generator will generate it correctly.

As long as this doesn't cause issues on the other platforms, this is a non-issue.

File: sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs (hunk 4 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:71]: fixed.

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs (hunk 5 above)
Comment (kasobol-msft) [reply sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:64]: same for datalake .

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:64]: I think it's unnecessary for Data Lake.  Data Lake calls into the BlockBlobClient.Query()/QueryAsync(), and follows this exact code path that was covered by the test I just wrote for Blobs.

Comment (kasobol-msft) [reply sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:64]: Why does code path like this exist in sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs ?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:64]: TODO write this test.

File: sdk/storage/Azure.Storage.Blobs/src/autorest.md (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/autorest.md]: fixed.

File: sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs (hunk 5 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/QuickQueryExtensions.cs:64]: fixed.

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**